### ISSUE TITLE
Support for different annotation file format, such as Pascal VOC

### ISSUE THREAD

**Issue by Sonyoyo** [EscVM/OIDv4_ToolKit#16]
Hello, you tool is great and very useful. 
I would like to use the downloaded dataset with tensorflow so I have to build a TFRecord file from it. However, before start training, I think I am going to modify/add some labels via the labelImg tool so I firstly need to build xml files in Pascal VOC format. It looks like it is not that hard to obtain them from your txt files but I am wondering whether you consider including this file format as output of you toolkit (I am a ML newbie so I hope I am not asking something obvious...).

**Comment by nilsfed** [comment]
Hey @Sonyoyo,

I would like to use Tensorflow aswell with an OIDv4 Subset. It would be great if there was support for Pascal VOC and especially TFRecord. 
So far I have been looking for a solution and experienced:

The .txt files generated by this ToolKit are already very similar to the YOLO format  

- **OIDv4:** `class leftX      topY        rightX bottomY` 
 

- **YOLO:** `class  x-center  y-center  width   height`

One could use a Python Script to generate YOLO format, which labelImg also supports, so you can modify and add your labels. I also came across a Repository where you can generate TFRecords from YOLO: [https://github.com/AlessioTonioni/tf-objdetector](https://github.com/AlessioTonioni/tf-objdetector), but I did not test it personally yet.

I hope that this can help you with your problem.


@EscVM
You created a great ToolKit, thank you!
Do you intend to support other bounding box annotation formats besides the current one? **Pascal VOC** and maybe a function to generate **TFRecords** would be great. I am going to research this topic more and would be willing to help and contribute.

**Comment by EscVM** [comment]
Hi @Sonyoyo 'n' @nilsfed!

Thank you! I really appreciate it.

I totally agree, it'd be a great idea to support also different types/formats of bounding boxes and directly download all the classes in a ready to use package! However, when I've started to realize how many different types of bb there could have been I thought that the best way was to download everything with an easy format to parse. So, simply if you want to have a specific format just write it down a program that transforms all the txt files in your desired one.

Here you can find a simple example I wrote to get yolo labels from oid ones ----> https://gist.github.com/EscVM/b6e5d60343c88f358742aa9e0de2cc3f

If you write other file like my one of other common format (Pascal, COCO...)  I could merge everything in the ToolKit;)

**Comment by Sonyoyo** [comment]
Hello,

thanks for your answers. I finally ended up creating a script to convert annotation files to PASCAL VOC format: the first test I did seems to work and I can see boxes and navigate through images in labelImg (which was my goal).
There I should be able to convert to TFRecord via the scripts provided by tensorflow itself.

@EscVM : if you think that my script could be useful for other and can be integrated to your toolkit, I can provide it. You could potentially add a kind of "--Format" key in your function call to choose among available output formats.

For your information, I was forced (by labelImg) to round the coordinate of boundaries to integer since they correspond to pixels (so float were not accepted).

**Comment by nilsfed** [comment]
Hey,

Today I did more or less the same as @Sonyoyo  already did :)
I wrote a Script to reach PASCAL VOC XML format, which also works with labelImg.
It checks all directories in the OID/Dataset folder and creates directories called To_PASCAL_XML right next to the original Label directories.

@EscVM : I hope you can use this script or some parts of it!  --> [gist link to my script](https://gist.github.com/nilsfed/1dbf1cf397db50c90705daa6a81a8dec)
If you have any questions, suggestions or need help, feel free to PM me or ask here, I think this is a great project and those annotation formats could help some people to modify/add labels and start training a model ;)

**Comment by martinbel** [comment]
It seems the conversion to VOC format is already available [here](https://github.com/cesarandreslopez/open-images-dataset-pascal-voc). 
Otherwise the process seems to be to get all bounding boxes for each image and the classes you care and save an xml file for each image.

**Comment by MitraTj** [comment]
Hi.
The tool created by you is great. 
Actually, I need to use pytorch visual relation detection with this dataset. And I want to know which strategy I need to follow? (download multiple classes at the same time creating separated folder and bounding boxes for each of them or download multiple classes and creating a common folder for all of them with a unique annotation file of each image).
thanks!

**Comment by keldrom** [comment]
@MitraTj is up to you this decision. It depends on how you want to train your network.

**Comment by keldrom** [comment]
Guys it seems that this issue is solved and so I'm going to close it. If you needs some other helps don't hesitate and contact us.
Thank you for the feedback and supports :D
